### DISCUSSION TITLE
Dynamic routing

### DISCUSSION THREAD

**Discussion by edazpotato** [vercel/next.js#22162] [Help] [answered]
Hi, I'm pretty new to next.js.
I just completed the tutorial abd I want to make an app with routes like this:
```
/ = Hompage or feed depending on if the user is sgined in
/whatever  = Various marketing pages
/[user-id] = A user profile page
/[user-id]/[post-id] = A post by that user
```
but I can't find out how to set this up from the docs. The homepage and marketing pages can be statically rendered because they won't change very often, but new users and posts will be created very often, so they can't be statically rendered.

Any help is appreciated!

**Comment by michrome** [comment, marked as answer]
Hi @edazpotato 

You're describing [dynamic routes](https://nextjs.org/docs/routing/dynamic-routes). Your project's file structure will start out looking something like the below, then you'll update [`getServerSideProps()`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) to fetch the data you need for the user profile and post pages.

```
/pages
  /index.js
  /whatever.js
  /[user-id].js
  /[user-id]/[post-id].js
```

```javascript
// pages/[user-id].js

const UserProfile = ({ userId }) => {
  return <p>Profile page for user: {userId}</p>;
};

export async function getServerSideProps({ query }) {
  return {
    props: { userId: query["user-id"] },
  };
}

export default UserProfile;
```

```javascript
// pages/[user-id]/[post-id].js

const UserPost = ({ postId, userId }) => {
  return (
    <p>
      This is post {postId} (by user {userId})
    </p>
  );
};

export async function getServerSideProps({ query }) {
  return {
    props: { postId: query["post-id"], userId: query["user-id"] },
  };
}

export default UserPost;
```

**Reply by edazpotato** [reply]
Thank you very much!

**Comment by stigkj** [comment]
A couple of clarifications in addition to the excellent answers from @michrome.

* Make sure that none of the names of your marketing pages are the same as any of your user ids, as non-dynamic routes will be matched before dynamic ones.
* You said that `/` should be statically rendered. Then I guess you have some code on the client that chooses whether the homepage or the feed is shown?

**Reply by edazpotato** [reply]
> Make sure that none of the names of your marketing pages are the same as any of your user ids, as non-dynamic routes will be matched before dynamic ones.

Yep, I'll make sure to do that. Thanks for the heads up

> You said that / should be statically rendered. Then I guess you have some code on the client that chooses whether the homepage or the feed is shown?

Yes, that's correct
